freqtrade/strategy/interface.py

```diff
@@ -200,12 +201,15 @@ def stop_loss_reached(self, current_rate: float, trade: Trade, current_time: dat
             # check if we have a special stop loss for positive condition
             # and if profit is positive
             stop_loss_value = self.stoploss
-            if 'trailing_stop_positive' in self.config and current_profit > 0:
+            sl_offset = self.config.get('trailing_stop_positive_offset', 0.0)
+
+            if 'trailing_stop_positive' in self.config and current_profit > sl_offset:
 
                 # Ignore mypy error check in configuration that this is a float
                 stop_loss_value = self.config.get('trailing_stop_positive')  # type: ignore
```

Comment:
yes and no, unfortunately i gave this quite some thought already.

yes it's overwriting it, the problem is that self.config.get() returns "Any" ... as everything could be in there, it's just a (patient) dict.

would i use `stop_loss_value = float(self.config.get('trailing_stop_positive')) ` it would in theory work (but that fails as float can't cast `Any` to `float` ...)

the problem is - L214 needs to use the "old" stoploss if this is not applying, and the "new" value if it is. the problem is therefore just "moved" to L214, which would fail with `error: Argument 2 to "adjust_stop_loss" of "Trade" has incompatible type "Optional[Any]"; expected "float"`.

i don't see a point in implementing workarounds just to get mypy to realize that json_validate already validated that this is in fact a float.

happy to get suggestions - my personal preferrence would be to have a class instead of the config dict - but that's a major change i would rather not include into a feature branch...

Reply:
no point of using too much extra time to get mypy happy 👍 

> my personal preferrence would be to have a class instead of the config dict
@xmatthias I think the new dataclass would fit the bill just right:
https://docs.python.org/3/library/dataclasses.html

Reply:
absolutely, but needs 3.7 ... which fails with an error on the ABC class